Source/GRChomboCore/GRAMRLevel.cpp

```diff
@@ -100,17 +100,25 @@ Real GRAMRLevel::advance()
 {
     CH_TIME("GRAMRLevel::advance");
 
-    // Work out roughly how fast the evolution is going since restart
-    double speed = (m_time - m_restart_time) / m_gr_amr.get_walltime();
-
-    // Get information on number of boxes on this level (helps with better load
-    // balancing)
-    const DisjointBoxLayout &level_domain = m_state_new.disjointBoxLayout();
-    int nbox = level_domain.dataIterator().size();
-    int total_nbox = level_domain.size();
-    pout() << "GRAMRLevel::advance level " << m_level << " at time " << m_time
-           << " (" << speed << " M/hr)"
-           << ". Boxes on this rank: " << nbox << " / " << total_nbox << endl;
+    // if 'print_progress_only_to_rank_0', still print if it's level 0 or
+    // t=restart_time
+    if (!m_p.print_progress_only_to_rank_0 || (procID() == 0) ||
+        m_time == m_restart_time)
+    {
+        // Work out roughly how fast the evolution is going since restart
+        double speed = (m_time - m_restart_time) / m_gr_amr.get_walltime();
+
+        // Get information on number of boxes on this level (helps with better
+        // load balancing)
+        const DisjointBoxLayout &level_domain = m_state_new.disjointBoxLayout();
+        int nbox = level_domain.dataIterator().size();
+        int total_nbox = level_domain.size();
+
+        pout() << "GRAMRLevel::advance level " << m_level << " at time "
+               << m_time << " (" << speed << " M/hr)"
+               << ". Boxes on this rank: " << nbox << " / " << total_nbox
+               << endl;
```

Comment:
Perhaps we can make this a private member function of `GRAMRLevel` to avoid duplicating code in `postRegrid`?